plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Instance/Flat.hs

```diff
@@ -116,7 +118,58 @@ instance ( Closed uni
         Error    ann      -> getSize ann
         Builtin  ann bn   -> getSize ann + getSize bn
 
-instance ( Closed uni
+-- This is a shorter encoding when the 'name' binders are DeBruijn indices; it omits the DeBruij index at the LamAbs node since they
+-- are always '0'.
+-- We use an overlapping instance as a workaround for not having to change the LamABS AST node.
+instance {-# OVERLAPPING #-} ( Closed uni
+         , uni `Everywhere` Flat
+         , Flat fun
+         , Flat ann
+         ) => Flat (Term DeBruijn uni fun ann) where
+    encode = \case
+        Var      ann n    -> encodeTerm 0 <> encode ann <> encode n
+        Delay    ann t    -> encodeTerm 1 <> encode ann <> encode t
+        LamAbs   ann _ t  -> encodeTerm 2 <> encode ann <> encode t
+        Apply    ann t t' -> encodeTerm 3 <> encode ann <> encode t <> encode t'
+        Constant ann c    -> encodeTerm 4 <> encode ann <> encode c
+        Force    ann t    -> encodeTerm 5 <> encode ann <> encode t
+        Error    ann      -> encodeTerm 6 <> encode ann
+        Builtin  ann bn   -> encodeTerm 7 <> encode ann <> encode bn
+
+    decode = go =<< decodeTerm
+        where go 0 = Var      <$> decode <*> decode
+              go 1 = Delay    <$> decode <*> decode
+              go 2 = flip LamAbs (DeBruijn (Index 0)) <$> decode <*> decode
+              go 3 = Apply    <$> decode <*> decode <*> decode
+              go 4 = Constant <$> decode <*> decode
+              go 5 = Force    <$> decode <*> decode
+              go 6 = Error    <$> decode
+              go 7 = Builtin  <$> decode <*> decode
+              go _ = fail "Failed to decode Term Name ()"
```

Comment:
error mentions the wrong type! It's not using `Name` and it's not necessarily using `()` as the annotation! Same problem with the decoder above, too.